lib/multiaddr.js

```diff
@@ -3,13 +3,32 @@
  * @returns {string} Parsed URI, e.g. `http://127.0.0.1:80`
  */
 export function multiaddrToHttpUrl (addr) {
-  const [, hostType, hostValue, ipProtocol, port, scheme, ...rest] = addr.split('/')
+  const [multiAddr, httpPathMultiAddr] = addr.split('/http-path')
+  const [, hostType, hostValue, ...multiAddrParts] = multiAddr.split('/')
+  let scheme, path, rest, port
+  if (addr.includes('/http-path')) {
+    [scheme, ...rest] = multiAddrParts
+    try {
+      // Remove leading slash and parse URI-encoded path
+      // If the http-path is empty, both `.../http-path/` and `.../http-path` are valid.
+      // See https://github.com/multiformats/multiaddr/blob/cab92e8e6da2e70c5f1b8aa59976e71e6922b392/protocols/http-path.md#usage
+      path = decodeURIComponent(httpPathMultiAddr.startsWith('/') ? httpPathMultiAddr.substring(1) : httpPathMultiAddr)
```

Comment:
```suggestion
      path = decodeURIComponent(httpPathMultiAddr.substring(1))
```

I changed the line to this, and tests are still all passing. Could you please find a test case where this would fail, or otherwise simplify as I have here?

Reply:
If we only call `httpPathMultiAddr.substring(1)` then we always remove the first character. If the first character is not `/` that would mean for example `examplePath/...` becomes `xamplePath/...`. I don't think this is what we want. 
I'll make sure this is included in the tests.